unit_tests/test_tester.py

```diff
@@ -196,6 +199,30 @@ def _validate_results(self):
         assert self.final_event.test_status == 'FAILED'
 
 
+class CriticalErrorNotCaught(ClusterTesterForTests):
+    @staticmethod
+    def test():
+        try:
+            ClusterHealthValidatorEvent.NodeStatus(
+                node='node-1',
+                message='Failed by some reason',
+                error='Reason to fail',
+                severity=Severity.CRITICAL,
+            ).publish()
+            end_time = time.time() + 2
+            while time.time() < end_time:
+                time.sleep(0.1)
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+    def _validate_results(self):
+        super()._validate_results()
+        # While running from pycharm and from hydra run-test exception inside subTest won't stop the test,
+        #  under hydra unit_test it stops running it and you don't see exception from next subtest.
+        assert 'ClusterHealthValidatorEvent' in self.events['CRITICAL'][0]
```

Comment:
Need to add one more assertion before this one checking that `self.events['CRITICAL']` is not empty.

Reply:
done